src/libs/actions/Report.js

```diff
@@ -2004,18 +2002,19 @@ function openReportFromDeepLink(url, isAuthenticated) {
 
     // Navigate to the report after sign-in/sign-up.
     InteractionManager.runAfterInteractions(() => {
-        SidebarUtils.isSidebarLoadedReady().then(() => {
-            if (route === ROUTES.CONCIERGE) {
-                navigateToConciergeChat(true);
-                return;
-            }
-            if (Session.isAnonymousUser() && !Session.canAccessRouteByAnonymousUser(route)) {
-                Navigation.isNavigationReady().then(() => {
+        Session.waitForUserSignIn().then(() => {
+            Navigation.waitForProtectedRoutes().then(() => {
```

Comment:
This wait promise callback although it solves the original issue, it revealed a hidden bug https://github.com/Expensify/App/issues/33001.